src/rmp/src/Restructure.cpp

```diff
@@ -219,15 +220,28 @@ void Restructure::runABC()
       Abc_Start();
       Abc_Frame_t* abc_frame = Abc_FrameGetGlobalFrame();
       const std::string command = "source " + abc_script_file;
+
+      fflush(stdout);
+      int stdout_fd = dup(STDOUT_FILENO);
+      const std::string abc_log_name = logfile_ + std::to_string(curr_mode_idx);
+      std::ofstream{abc_log_name.c_str()};
+      int redir_fd = open(abc_log_name.c_str(), O_WRONLY);
+      dup2(redir_fd, STDOUT_FILENO);
+      close(redir_fd);
```

Comment:
You could make a RAII class to handle the redirect/reset.  It would ensure that the output get reset in case of any exception (probably abc doesn't throw but who knows).